arangod/RocksDBEngine/RocksDBHotBackup.cpp

```diff
@@ -767,4 +780,95 @@ bool RocksDBHotBackupRestore::createRestoringDirectory(std::string & restoreDirO
 } // RocksDBHotBackupRestore::createRestoringDirectory
 
 
+////////////////////////////////////////////////////////////////////////////////
+/// @brief RocksDBHotBackupList
+///        POST:  Returns array of Hotbackup directory names
+////////////////////////////////////////////////////////////////////////////////
+RocksDBHotBackupList::RocksDBHotBackupList(const VPackSlice body)
+  : RocksDBHotBackup(body)
+{
+}
+
+
+RocksDBHotBackupList::~RocksDBHotBackupList() {
+}
+
+
+void RocksDBHotBackupList::parseParameters(rest::RequestType const type) {
+
+  _valid = (rest::RequestType::POST == type);
+
+  if (!_valid) {
+    _result.add(VPackValue(VPackValueType::Object));
+    _result.add("httpMethod", VPackValue("only POST allowed"));
+  } // if
+
+  if (!_valid) {
```

Comment:
This does not look sensible with two identical if statements after another.

Reply:
fixed.  vpack stuff also wrapped in try/catch